### PR TITLE
feat(cb2-10457): support ADR cert flow

### PR DESCRIPTION
**Pull request by naathanbrown**
## Ticket title
ADR cert flow
Also contains moving types to V3 so some type narrowing required on the plates and letters
[10457](https://dvsa.atlassian.net/browse/CB2-10457)

## Checklist
- [ ] Code has been tested manually
- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/models/adrPassCertificate.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    // S3 metadata
+    this.metaData.vin = techRecord.vin;
+    this.metaData['should-email-certificate'] = 'false';
+
+    const adrData: AdrCert = {
+      // ADR data
+      vin: techRecord.vin,
+      make: techRecord.techRecord_make,
+      vrm: techRecord.techRecord_vehicleType === 'hgv' ? techRecord.primaryVrm : techRecord.trailerId,
```
Comment (owen-corrigan) [inline src/models/adrPassCertificate.ts]:
```suggestion
      vrm: techRecord.techRecord_vehicleType === 'trl' ? techRecord.trailerId : techRecord.primaryVrm,
```
I think we need the ternary this way round to support lgv?

Comment (naathanbrown) [reply src/models/adrPassCertificate.ts]: yeah just did this, lol

File: src/models/ministryPlate.ts [hunk 2]
```diff
@@ -123,12 +139,12 @@ export class MinistryPlateDocument extends DocumentModel {
     this.Reissue = { Reason: plate.plateReasonForIssue };
 
     // S3 metadata
-    this.metaData.vrm = techRecord.primaryVrm ?? techRecord.trailerId;
+    this.metaData.vrm = techRecord.techRecord_vehicleType === 'hgv' ? techRecord.primaryVrm : techRecord.trailerId;
```
Comment (pbardy2000) [inline src/models/ministryPlate.ts]: Think you need to invert the ternary here as well:
```suggestion
    this.metaData.vrm = techRecord.techRecord_vehicleType === 'trl' ? techRecord.trailerId : techRecord.primaryVrm;
```

**Review by owen-corrigan — APPROVED** [review]

File: src/models/ministryPlate.ts (hunk 2 above)
Comment (owen-corrigan) [reply src/models/ministryPlate.ts]: With this being plates I'm not sure it does right? I think they're only applicable to HGV or TRL

File: src/models/ministryPlate.ts [hunk 3]
```diff
@@ -79,7 +81,7 @@ export class MinistryPlateDocument extends DocumentModel {
     const plateData: Partial<MinistryPlate> = {
       plateSerialNumber: plate.plateSerialNumber,
       dtpNumber: techRecord.techRecord_brakes_dtpNumber,
-      primaryVrm: techRecord.primaryVrm ?? techRecord.trailerId,
+      primaryVrm: techRecord.techRecord_vehicleType === 'hgv' ? techRecord.primaryVrm : techRecord.trailerId,
```
Comment (shivangidas) [inline src/models/ministryPlate.ts]:
```suggestion
      primaryVrm: techRecord.techRecord_vehicleType === 'trl' ? techRecord.trailerId : techRecord.primaryVrm,
```

File: src/models/request.ts [hunk 4]
```diff
@@ -23,3 +25,5 @@ export interface Letter {
   letterIssuer: string;
   letterDateRequested: string;
 }
+
+export type HgvOrTrl = TechRecordType<'hgv', 'get'> | TechRecordType<'trl', 'get'>;
```
Comment (shivangidas) [inline src/models/request.ts]: guessing this needs to include lgv too

File: src/models/ministryPlate.ts (hunk 2 above)
Comment (naathanbrown) [reply src/models/ministryPlate.ts]: Yes, this is written badly but I'm not in the mood or have the time to improve it!

File: src/models/ministryPlate.ts (hunk 3 above)
Comment (naathanbrown) [reply src/models/ministryPlate.ts]: Not touching plates more that I need too, sorry.

**Review by pbardy2000 — APPROVED** [review]

**Review by shivangidas — APPROVED** [review]

**Pull request merged**